File: src/main/java/shopping/wish/WishRepository.java
package shopping.wish;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface WishRepository extends JpaRepository<Wish, Long> {
    Page<Wish> findByMemberId(Long memberId, Pageable pageable);

    Optional<Wish> findByMemberIdAndProductId(Long memberId, Long productId);
}

File: src/main/kotlin/shopping/domain/wish/WishRepository.kt
package shopping.domain.wish

interface WishRepository {
    fun findByMemberId(memberId: Long): List<Wish>

    fun findByMemberIdAndProductId(
        memberId: Long,
        productId: Long,
    ): Wish?

    fun existsByMemberIdAndProductId(
        memberId: Long,
        productId: Long,
    ): Boolean

    fun save(wish: Wish): Wish

    fun deleteById(id: Long)
}
